test/system-contracts/hedera-account-service/crypto-allowance/cryptoAllowance.js

```diff
@@ -302,4 +308,95 @@ describe('@CryptoAllowance Test Suite', () => {
+
+    const correctSignerReceipt = await (
+      await cryptoAllowanceContract.isAuthorizedRawPublic(
+        walletB.address, // correct signer
+        messageHash,
+        signature,
+        Constants.GAS_LIMIT_1_000_000
+      )
+    ).wait();
+
+    const correctSignerReceiptResponseCode = correctSignerReceipt.logs.find(
+      (l) => l.fragment.name === 'ResponseCode'
+    ).args[0];
+
+    const correctSignerReceiptResponse = correctSignerReceipt.logs.find(
+      (l) => l.fragment.name === 'IsAuthorizedRaw'
+    ).args;
+
+    expect(correctSignerReceiptResponseCode).to.eq(22n);
+    expect(correctSignerReceiptResponse[0]).to.eq(walletB.address);
+    expect(correctSignerReceiptResponse[1]).to.be.true;
+
+    const incorrectSignerReceipt = await (
+      await cryptoAllowanceContract.isAuthorizedRawPublic(
+        walletC.address, // incorrect signer
+        messageHash,
+        signature,
+        Constants.GAS_LIMIT_1_000_000
+      )
+    ).wait();
+
+    const incorrectSignerReceiptResponseCode = incorrectSignerReceipt.logs.find(
+      (l) => l.fragment.name === 'ResponseCode'
+    ).args[0];
+
+    const incorrectSignerReceiptResponse = incorrectSignerReceipt.logs.find(
+      (l) => l.fragment.name === 'IsAuthorizedRaw'
+    ).args;
+
+    expect(incorrectSignerReceiptResponseCode).to.eq(22n);
+    expect(incorrectSignerReceiptResponse[0]).to.eq(walletC.address);
+    expect(incorrectSignerReceiptResponse[1]).to.be.false;
+  });
+
+  it('Should verify message signature using isAuthorizedRawPublic for ED25519 account', async () => {
+    const newEdPK = PrivateKey.generateED25519();
+    const messageHash = Buffer.from(messageToSign);
+    const signature = `0x${Buffer.from(newEdPK.sign(messageHash)).toString('hex')}`;
+
+    const newEdPubKey = newEdPK.publicKey;
+    const client = await Utils.createSDKClient();
+
+    const edSignerAccount = (
+      await (
+        await new AccountCreateTransaction()
+          .setKey(newEdPubKey)
+          .setInitialBalance(Hbar.fromTinybars(1000))
+          .execute(client)
+      ).getReceipt(client)
+    ).accountId;
+
+    const signerAlias = `0x${edSignerAccount.toSolidityAddress()}`;
+
+    const correctSignerReceipt = await (
+      await cryptoAllowanceContract.isAuthorizedRawPublic(
+        signerAlias,
+        messageHash,
+        signature,
+        Constants.GAS_LIMIT_10_000_000
+      )
+    ).wait();
+
+    const correctSignerReceiptResponseCode = correctSignerReceipt.logs.find(
+      (l) => l.fragment.name === 'ResponseCode'
+    ).args[0];
+
+    const correctSignerReceiptResponse = correctSignerReceipt.logs.find(
+      (l) => l.fragment.name === 'IsAuthorizedRaw'
+    ).args;
+
+    expect(correctSignerReceiptResponseCode).to.eq(22n);
+    expect(correctSignerReceiptResponse[0].toLowerCase()).to.eq(
+      signerAlias.toLowerCase()
+    );
+    expect(correctSignerReceiptResponse[1]).to.be.true;
+  });
```

Comment:
Maybe there could be one or two negative case tests?

Reply:
Ah yeah I grouped the negative case within the one that used ECDSA key https://github.com/hashgraph/hedera-smart-contracts/pull/997/files#diff-90094514c9235785c54135c79d4a4ea158720b3ac9c2ba4ca0a175b5aa0ffe8eR338-R357. 

But yeah let me separate it out and add another case for ED key.

Reply:
Updated